File: rsa.c
#define _GNU_SOURCE
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int is_prime(unsigned long long int n)
{
	int i, flag = 1;
	
	for (i = 2; i <= sqrt(n); i++)
	{
		/* If n is divisible by any number between*/
		/* 2 and n/2, it is not prime*/
		if (n % i == 0) {
			flag = 0;
			break;
		}
	}

	if (n <= 1)
		flag = 0;

	return (flag);
}

unsigned long long *factorize(long n)
{
	long long i;
	static unsigned long long r[2];

	for (i = 2; i < n; i++)
	{
		if (n % i == 0)
		{
			if (is_prime((unsigned long long)i) && is_prime((unsigned long long)n / i))
			{
				r[0] = (unsigned long long)i;
				r[1] = (unsigned long long)n / i;
				return (r);
			}
		}
	}

	r[0] = 1;
	r[1] = (unsigned long long)n;
	return (r);
}


int main(int argc, char *argv[])
{
	FILE *fp;
	char *line = NULL, *ptr;
	size_t len = 0;
	ssize_t read;
	unsigned long long *factors, number;

	(void) factors;

	if (argc != 2)
	{
		printf("Usage: %s <file>", argv[0]);
		exit(1);
	}

	fp = fopen(argv[1], "r");
	if (fp == NULL)
		exit(EXIT_FAILURE);

	while ((read = getline(&line, &len, fp)) != -1)
	{
		number = strtoull(line, &ptr, 10);
		if (number <= 18446744073709551615)
		{
			factors = factorize(number);
			printf("%llu=%llu*%llu\n", number, factors[1], factors[0]);
		}
	}

	fclose(fp);
	if (line)
		free(line);
	exit(EXIT_SUCCESS);
}

File: rsa.py
#!/usr/bin/python3
import sys
from math import sqrt


def is_prime(n):
	flag = 1
	for i in range(2, int(sqrt(n))):
		#/* If n is divisible by any number between*/
		#/* 2 and n/2, it is not prime*/
		if n % i == 0:
			flag = 0
			break

	if n <= 1:
		flag = 0

	return (flag)


def factorize(number):
	for i in range(2, number):
		if number % i == 0:
			if is_prime(i) and is_prime(int(number / i)):
				print("{:d}={:d}*{:d}".format(number, int(number / i), i))
				return
	print("{:d}={:d}*{:d}".format(number, number, 1))


if __name__ == "__main__":
	if len(sys.argv) == 2:
		filename = sys.argv[1]
		with open(filename, 'r') as file:
			for line in file.readlines():
				try:
					num = int(line.strip())
					factorize(num)
				except:
					pass
	else:
		sys.exit("Usage: {} <file>".format(sys.argv[0]))
